pallets/pallet-deposit-storage/src/fungible/mod.rs

```diff
@@ -88,7 +91,7 @@ impl<Namespace, Key> From<PalletDepositStorageReason<Namespace, Key>> for HoldRe
 	fn from(_value: PalletDepositStorageReason<Namespace, Key>) -> Self {
 		// All the deposits ever taken like this will count towards the same hold
 		// reason.
-		Self::Deposit
+		Self::FungibleImpl
```

Comment:
This helps keeping the two functionalities of "regular" deposit and "system" deposits completely separated from each other, and having two different reasons also allows us to write much more accurate and strict `try-state` checks.